source/PlayerInfo.cpp

```diff
@@ -1134,26 +1140,31 @@ void PlayerInfo::AddShip(const shared_ptr<Ship> &ship)
 
 
 // Adds a ship of the given model with the given name to the player's fleet.
-void PlayerInfo::BuyShip(const Ship *model, const string &name)
+void PlayerInfo::BuyShip(const Ship *model, const string &name, const bool isFromShop)
 {
 	if(!model)
 		return;
 
-	int day = date.DaysSinceEpoch();
-	int64_t cost = stockDepreciation.Value(*model, day);
-	if(accounts.Credits() >= cost)
+	const Ship *newShip = isFromShop ? model : StockShip(model);
```

Comment:
it's a small thing but do we need a new variable?
what I mean is that we dont use model after this so we could just update model if needed

Reply:
We do use model after this, on line 1167.